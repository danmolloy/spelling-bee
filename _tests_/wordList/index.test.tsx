import { render, screen, fireEvent, act } from "@testing-library/react";
import "@testing-library/jest-dom"
import WordList, { capitalize } from "../../components/wordList";

const mockProps = {
  words: ["Car", "Bar", "Mat", "Door", "Hat", "Dog"]
}

describe("WordList component", () => {
  beforeEach(() => {
    render(<WordList {...mockProps} />)
  })
  it("Renders", () => {
    const wordList = screen.getByTestId("word-list")
    expect(wordList).toBeInTheDocument()
  })
  it("ArrowIcon is in the document", () => {
    const arrowIcon = screen.getByTestId("arrow-icon")
    expect(arrowIcon).toBeInTheDocument()
  })
  it("List preview is default view", () => {
    const listPreview = screen.getByTestId("list-preview")
    expect(listPreview).toBeInTheDocument()
  })
  it("Shows/hides FullList on ArrowIcon click", () => {
    const arrowIcon = screen.getByTestId("arrow-icon")
    const listPreview = screen.getByTestId("list-preview")
    expect(listPreview).toBeInTheDocument()
    act(() => {
      fireEvent.click(arrowIcon)
    })
    const fullList = screen.getByTestId("full-list-div")
    expect(fullList).toBeInTheDocument()
    expect(listPreview).not.toBeInTheDocument()
  })
})

describe("Capitalize function", () => {
  it("Capitalizes words", () => {
    expect(capitalize("iron")).toMatch(/^Iron$/)
    expect(capitalize("IRON")).toMatch(/^Iron$/)
    expect(capitalize("Iron")).toMatch(/^Iron$/)
    expect(capitalize("iRON")).toMatch(/^Iron$/)
  })
})