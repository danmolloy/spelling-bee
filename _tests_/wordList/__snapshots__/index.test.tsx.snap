// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WordList /> matches snapshot 1`] = `
<div
  class="w-full md:w-2/5 bg-white absolute z-30 border rounded-lg p-3 flex flex-row justify-between items-center font-light mt-10 transition-all duration-300"
  data-testid="word-list"
>
  <div
    class=" w-full flex flex-row "
  >
    <div
      class="w-full flex flex-row flex-wrap transition-all duration-500"
      data-testid="list-preview"
    >
      <p
        class="px-1"
      >
        Cat
      </p>
      <p
        class="px-1"
      >
        Mat
      </p>
    </div>
    <div
      class="w-8 h-8 flex justify-center items-center self-start rounded-full hover:bg-gray-100 active:bg-gray-200"
      data-testid="arrow-icon"
    >
      <div
        class="border-black border-r-2 border-b-2 h-3 w-3 self-end rotate-45 mb-3 "
      />
    </div>
  </div>
</div>
`;
